### PR TITLE
Chore: modifies funding logic

### PR DESCRIPTION
**Pull request by fabianschu**
### Changes: split sponsor (who adds funding) and workflow admin + add deployer allowlist:
* previously the sponsor was equal to the workflow admin => for legal reasons this needs to be changed
* additionally: anyone could use a funding for a deployment, deploying with an arbitrary (undesired) config

=> both is addressed in this PR

---

* new logic: sponsor (who adds funding) specifies 
  * which deployer can deploy a workflow 
  * for which beneficiary (receives initial issuance supply and CURVE_INTERACTION as well as PAYMENT_PUSHER role)
  * with which admin (is the workflow admin)
* `createPIMWorkflow` interface stays unchanged

### CODE REVIEW

File: test/factories/custom/Restricted_PIM_Factory_v1.t.sol [hunk 1]
```diff
@@ -122,112 +122,336 @@ contract Restricted_PIM_Factory_v1Test is E2ETest {
...
-        |       └── then it grants payment pusher role to beneficiary
-        |       └── then it transfers ownership of mint wrapper to admin
-        |       └── then it renounces ownership of issuance token
-        |       └── then it revokes orchestrator admin rights from factory and transfers them to admin
-        |       └── then it emits a PIMWorkflowCreated event
-        └── given that there is no funding
-        |   └── when called
-        |       └── then it reverts
-    */
+    // /* Test createPIMWorkflow
```
Comment (0xNuggan) [inline test/factories/custom/Restricted_PIM_Factory_v1.t.sol]: Why this change?

**Review by 0xNuggan — CHANGES_REQUESTED** [review]
Looks good, only the one style thing

File: test/factories/custom/Restricted_PIM_Factory_v1.t.sol (hunk 1 above)
Comment (fabianschu) [reply test/factories/custom/Restricted_PIM_Factory_v1.t.sol]: makes no sense, ty

File: src/factories/custom/Restricted_PIM_Factory_v1.sol [hunk 2]
```diff
@@ -125,37 +131,65 @@ contract Restricted_PIM_Factory_v1 is
...
-    {
+    function addFunding(
+        address deployer,
+        address beneficiary,
+        address admin,
+        address token,
+        uint amount
+    ) external {
+        Funding storage funding = fundings[deployer][beneficiary][admin][token];
+        if (funding.sponsor != address(0) && funding.sponsor != _msgSender()) {
```
Comment (ben-kaufman) [inline src/factories/custom/Restricted_PIM_Factory_v1.sol:142]: This makes it possible to front run calls to this with 1 devimal of of the token and make adding funding fail

Comment (omega-audits) [inline src/factories/custom/Restricted_PIM_Factory_v1.sol]: This makes it possible to front run calls to this with 1 decimal of of the token and make adding funding fail
